File: C/424.c
#include <stdio.h>
#include <stdlib.h>

// 424 - Ahorro Infantil - Iván - https://github.com/wildfireOfMine

int main()
{
    int loop;
    scanf("%d", &loop);
    while (loop != 0) {
        int i;
        int result = 0;
        int maxResult = 0;
        for (i = 0; i<loop; i++) {
            int n = 0;
            scanf("%d", &n);
            result += n;
            if (result > maxResult) {
                maxResult = result;
            }
        }
        printf("%d %d\n", result, maxResult);
        scanf("%d", &loop);
    }
    return 0;
}

File: Java/424.java
import java.util.Scanner;

/**
 *
 * 424 - Ahorro Infantil - Iván - https://github.com/wildfireOfMine
 */
public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int prueba = sc.nextInt();
        while (prueba != 0) {
            int gasto = 0;
            int mayor = 0;
            for (int i = 1; i <= prueba; i++) {
                int ahorro = sc.nextInt();
                gasto += ahorro;
                if (mayor < gasto) {
                    mayor = gasto;
                }
            }
            System.out.println(gasto + " " + mayor);
            prueba = sc.nextInt();
        }
    }

}
